crates/iota-data-ingestion-core/README.md

```diff
@@ -0,0 +1,352 @@
+- The third branch is responsible for distributing the incoming checkpoints from `CheckpointReader` across all Workers Pools.
+
+## Use cases
+
+The use cases for the `iota-data-ingestion-core` can be endless. It not only could be used to create Custom Indexers but also for analyzing the data and integrate it with external systems
+
+#### 1. **Building Custom Indexers for Specific Data:**
+
+- **Indexing Only the Necessary Data:** Depending on the specific use case, a custom indexer can be developed to store only the relevant data. For example, businesses can register transactions based on the sender and recipient, allowing them to retain essential transactional data for the parties they need without the concern that a full node may delete this information.
+- **Indexer as a service**: Provide users with the tools to efficiently index and query specific data, unlocking the potential of the IOTA Tangle for various applications.
+
+#### 2. **Data Analysis and Visualization:**
+
+- **Historical Analysis:** Analyzing historical data to identify trends, patterns, and anomalies.
+- **Real-time Analytics:** Processing data in real-time to generate insights and alerts.
+- **Data Visualization:** Creating visualizations to represent data insights in a user-friendly way.
+
+#### 3. **Integration with External Systems:**
+
+- **Data Pipelines:** Integrating with other data processing systems (e.g., Apache Kafka, Apache Spark) for large-scale data processing.
+- **Machine Learning:** Feeding indexed data into machine learning models for predictive analytics.
+- **Alerting Systems:** Triggering alerts based on specific events or conditions identified in the data.
+
+## Usage in existing crates
+
+As an example of a custom indexer the `iota-data-ingestion-core` is used in the `iota-indexer` and `iota-analytics-indexer` both persist data in different storages with completely different use cases:
+
+- **iota-indexer**: The primary function of the indexer is to provide historical data from the ledger, which is crucial for both Explorer and developers who want to utilize ledger data in their applications. By pre-processing and storing this data, the indexer significantly reduces the load on the full node, preventing performance degradation from frequent queries. The data is stored in `PostgreSQL`.
+
+- **iota-analytics-indexer**: This is a specialized indexer designed to analyze and store key metrics from the IOTA blockchain. It tracks network statistics like transaction volume and throughput, user activity such as the number of active addresses, and the execution of Move calls, providing insights into smart contract usage. The collected data is efficiently stored in Object Storage and subsequently transferred to columnar databases like BigQuery or Snowflake, optimized for analytical queries.
```

Comment:
Actually the analytics indexer is not performing any analysis of the data, It's just exporting all the data for analysis in other tools.

> It tracks network statistics like transaction volume and throughput, user activity such as the number of active addresses, and the execution of Move calls

Those are metrics currently computed by `iota-indexer`.

They potentially may be computed from the data exported by the analytical-indexer but it's not done currently.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/4570/commits/d4da5fa9970747c54848406c0d814bfbf3351fef)